frontend/script.js

```diff
@@ -0,0 +1,57 @@
+function showSideBar() {
+    const sidebar = document.querySelector('.hamburger');
+    sidebar.style.display = 'flex';
+}
+
+function hideSideBar() {
+    const sidebar = document.querySelector('.hamburger');
+    sidebar.style.display = 'none';
+}
+
+// Specify the API endpoint for user data
+const apiUrl = 'http://localhost:8080/products/';
+// Make a GET request using the Fetch API
+fetch(apiUrl)
+    .then((response) => {
+        if (!response.ok) {
+            throw new Error('Network response was not ok');
+        }
+        return response.json();
+    })
+    .then((data) => {
+        data.data.forEach((element) => {
+            if ('content' in document.createElement('template')) {
+                const trendingBody = document.querySelector('#trending');
+                const template = document.querySelector('#trending-product');
+                const clone = template.content.cloneNode(true);
+                let tp = clone.querySelectorAll('div');
+                const img = tp[0].querySelector('img');
+                img.addEventListener('error', function (event) {
+                    event.target.src = './assets/images/trending-product.webp';
+                    event.onerror = null;
```

Comment:
what's happening here and why is this needed? 😕

Reply:
Hi Joe,

This code snippet handles scenarios where an image fails to load on a webpage. It begins by selecting the first image element (img) found within the div (tp[0])

The addEventListener function is then used to listen for the error event on the selected image. If, for any reason, the image fails to load from the server or encounters an error during loading, this event will be triggered.

In response to the error event, a callback function is invoked. This function receives the event object as its argument. Inside the callback, it's ensured that the onerror event is nullified to prevent infinite loops in case of repeated errors.

The key action takes place within the callback: the src attribute of the problematic image (event.target.src) is reassigned to a default image link ('./assets/images/trending-product.webp'). This ensures that if the original image fails to load, a predetermined fallback image is displayed instead.

Reply:
Looking back at this, I may have overcomplicated this. I've updated the handleError function which just reassigns the source of the image to a default image.